tools/regtesting/do_regtest

```diff
@@ -10,7 +10,6 @@
 # command line argument passing
 #
 ###################################################################################
-export LC_ALL=en_US.UTF-8
```

Comment:
The point of this one here is to make sure that when a user reports an error it will be in `en_US.UTF-8` rather than whatever she has on her system. For example messages coming from other (possibly localized) runtimes.

Reply:
This is one is tricky. The `en_US.UTF-8` is not installed by default on Ubuntu. Hence, many of our users won't have it either. The obvious choice would be `C.UTF-8`, but that seems to have [portability issues]( https://github.com/cp2k/cp2k/commit/61740544b7b575d0d5daefac40fdf8df18c7049b) too. Since it's only about the output, maybe we should just go back to `LC_ALL=C`?

Reply:
```console
$ wget http://releases.ubuntu.com/18.10/ubuntu-18.10-desktop-amd64.manifest -q -O- | grep -i locales
krb5-locales	1.16-2ubuntu1
locales	2.28-0ubuntu1
```

I doubt many users work inside a minimal Ubuntu docker container.

Reply:
... and since SSH usually has `AcceptEnv LANG LC_ALL` a users locale (which is in todays times usually a UTF-8 locale) is also set when logged in from remote. Hence a proper locale rather than `C` or `C.UTF-8` should actually be closer to what our users have.

Reply:
My experience (with e.g. raspbian) is that distributions tend to provide a minimal set of locales (C locale and selected system language) plus the locale generator. So, they leave it to the user to generate missing, actually desired additional locales.

Reply:
> I doubt many users work inside a minimal Ubuntu docker container.

Certainly, every desktop users will have some locales installed, but I don't think we can assume that `en_US.UTF-8` is always among them.

So, I think we should just not overwrite the LC settings. I'm not even sure what localized error messages do_regtest would produce and we respect the user's settings everywhere else anyways.

Reply:
> My experience (with e.g. raspbian) is that distributions tend to provide a minimal set of locales (C locale and selected system language) plus the locale generator.

JeOS distros like raspbian definitely limit the number of locales installed due to space limitations on MicroSDs, I am not so sure about usual desktop distros.

> So, I think we should just not overwrite the LC settings. I'm not even sure what localized error messages do_regtest would produce and we respect the user's settings everywhere else anyways.

Error messages coming from a C++, Fortran or MPI runtime for example.

Reply:
> Error messages coming from a C++, Fortran or MPI runtime for example.

I can't recall ever seeing a localized error messages from any of those runtimes.

Nevertheless, `LC_ALL=C` ensures that messages are in English without making any further assumptions about the user.